resolvers/node/resolution.rs

```diff
@@ -363,11 +404,23 @@ impl<
       );
     }
 
-    if resolved.is_node_url() {
-      return Ok(resolved);
-    }
-
-    let path = resolved.into_path()?;
+    let (path, maybe_url) = match resolved {
+      LocalUrlOrPath::Url(url) => {
+        if url.scheme() == "file" {
+          (url_to_file_path(&url)?, Some(url))
+        } else {
+          return Ok(UrlOrPath::Url(url));
+        }
+      }
+      LocalUrlOrPath::Path(LocalPath { path, known_exists }) => {
+        if known_exists {
```

Comment:
As part of this change, we do the finalize_resolution step when resolving npm specifiers too (so that these better errors can happen and so the node resolution code is always using the same path). I was concerened about the extra fs hit that brings, so I updated the code to also keep track of whether we know a path exists in order to avoid the finalization check if the path is known to exist.